src/bytes.js

```diff
@@ -0,0 +1,1522 @@
+    var ret;
+    var i;
+    var num;
+    ret = "";
+    for (i = 0; i < this.v.byteLength; i++) {
+        num = this.v.getUint8(i);
+        if ((num < 9) || (num > 10 && num < 13) || (num > 13 && num < 32) || (num > 126)) {
+            ret += makehexform(num);
+        } else if (num === 9 || num === 10 || num === 13 || num === 92) {
+            switch (num) {
+                case 9:
+                    ret += "\\t";
+                    break;
+                case 10:
+                    ret += "\\n";
+                    break;
+                case 13:
+                    ret += "\\r";
+                    break;
+                case 92:
+                    ret += "\\\\";
+                    break;
+            }
+        } else {
+            ret += String.fromCharCode(num);
+        }
+    }
+    ret = "b'" + ret + "'";
+    return new Sk.builtin.str(ret);
+};
+
+Sk.builtin.bytes.prototype.mp$subscript = function (index) {
+    var ret;
+    var i;
+    if (Sk.misceval.isIndex(index)) {
+        i = Sk.misceval.asIndex(index);
+        if (i !== undefined) {
+            if (i < 0) {
+                i = this.v.byteLength + i;
+            }
+            if (i < 0 || i >= this.v.byteLength) {
+                throw new Sk.builtin.IndexError("index out of range");
+            }
+            return new Sk.builtin.int_(this.v.getUint8(i));
+        }
+    } else if (index instanceof Sk.builtin.slice) {
+        ret = [];
+        if (index.start.v < 0) {
+            index.start.v = 0;
+        }
+        if (index.stop.v > this.v.byteLength) {
+            index.stop.v = this.v.byteLength;
+        }
+        if (index.stop.v < 0) {
+            index.stop.v = this.v.byteLength + index.stop.v;
+        }
+        index.sssiter$(this, function (i, wrt) {
+            ret.push(new Sk.builtin.int_(wrt.v.getUint8(i)));
+        });
+        return new Sk.builtin.bytes(new Sk.builtin.list(ret));
+    }
+
+    throw new Sk.builtin.TypeError("byte indices must be integers, not " + Sk.abstr.typeName(index));
+};
+
+Sk.builtin.bytes.prototype.ob$eq = function (other) {
+    var i;
+    var val1;
+    var val2;
+    var iter1;
+    var iter2;
+
+    if (this === other) {
+        return Sk.builtin.bool.true$;
+    }
+
+    if (!(other instanceof Sk.builtin.bytes)) {
+        return Sk.builtin.bool.false$;
+    }
+    if (this.sq$length() != other.sq$length()) {
+        return Sk.builtin.bool.false$;
+    }
+    iter1 = this.tp$iter();
+    iter2 = other.tp$iter();
+
+    for (i = 0; i < this.sq$length(); i ++) {
+        val1 = iter1.tp$iternext();
+        val2 = iter2.tp$iternext();
+        if (val1 === undefined || val2 === undefined) {
+            break;
+        }
+        if (val1.v != val2.v) {
+            return Sk.builtin.bool.false$;
+        }
+    }
+
+    return Sk.builtin.bool.true$;
+};
+
+Sk.builtin.bytes.prototype.ob$ne = function (other) {
+    return (!(this.ob$eq(other)));
```

Comment:
this will always return false since `ob$eq` will return a `pyBool` I would suggest making `ob$eq` return js `boolean`s

Would also be good to implement the other comparisons
https://github.com/python/cpython/blob/46e19b61d31ba99f049258efa4ff1334856a3643/Lib/test/test_bytes.py#L223

Reply:
Changing the semantics of `ob$eq` is probably out of scope for this PR - we should make `bytes`'s implementation match the rest of the codebase for now!

Reply:
oops sorry yes - stick with the `pyBool` return values.